### PR TITLE
Sigma windows

### PR DESCRIPTION
**Pull request by jaegeral**
This PR will add capabilities to support Sigma Windows rules and improves the overall Sigma workflow.

There is a Doc that explains the Sigma approach, when running Sigma analyzer, it will create a story as well as an aggregation.

Please note that there is only a small amount of rules included yet, as we have not spoken about rule distribution.

The corresponding issue for that is: https://github.com/google/timesketch/issues/1199

### CODE REVIEW

**Comment by jaegeral** [comment]
Maybe not merge it yet, I might want to introduce some more features / coverage

**Comment by kiddinn** [comment]
is this ready for a review? or do you still want to add some more features?

**Comment by jaegeral** [comment]
Yes, this is ready for review. More features will come after that for sure.

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 1]
```diff
@@ -112,5 +166,28 @@ class LinuxRulesSigmaPlugin(SigmaPlugin):
...
+    NAME = 'sigma_windows'
+
+class TestRulesSigmaPlugin(SigmaPlugin):
+    """Sigma plugin to run Windows rules."""
+
+    _RULES_PATH = '../../../data/test_rules'
+
+    NAME = 'a_sigma_test'
+
+class PowershellRulesSigmaPlugin(SigmaPlugin):
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:183]: Do we want to put the powershell rules under windows instead?

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 2]
```diff
@@ -112,5 +166,28 @@ class LinuxRulesSigmaPlugin(SigmaPlugin):
...
+    _RULES_PATH = '../../../data/windows'
+
+    NAME = 'sigma_windows'
+
+class TestRulesSigmaPlugin(SigmaPlugin):
+    """Sigma plugin to run Windows rules."""
+
+    _RULES_PATH = '../../../data/test_rules'
+
+    NAME = 'a_sigma_test'
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:181]: s/a_sigma_test/sigma_test

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 3]
```diff
@@ -112,5 +166,28 @@ class LinuxRulesSigmaPlugin(SigmaPlugin):
...
+
+class PowershellRulesSigmaPlugin(SigmaPlugin):
+    """Sigma plugin to run Windows rules."""
+
+    _RULES_PATH = '../../../data/powershell'
+
+    NAME = 'a_sigma_powershell'
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:188]: s/a_sigma_powershell/sigma_powerhell

File: timesketch/lib/analyzers/sigma_tagger.py [hunk 4]
```diff
@@ -100,8 +120,42 @@ def run(self):
...
-            output_string += '* {0:s}: {1:d}'.format(
+            output_string += '* {0:s}: {1:d}\n'.format(
                 tag_name, number_of_tagged_events)
+
+        if simple_counter > 0:
+            view = self.sketch.add_view(
+                view_name='Sigma Rule matches', analyzer_name=self.NAME,
+                query_string='tag:"sigma*"',
+                additional_fields=self._FIELDS_TO_INCLUDE)
+            params = {
```
Comment (berggren) [inline timesketch/lib/analyzers/sigma_tagger.py:131]: agg_params

File: data/windows/win_susp_add_sid_history.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
+title: Addition of SID History to Active Directory Object
```
Comment (berggren) [inline data/windows/win_susp_add_sid_history.yml:1]: Can we add a reference to where this rule was copied from?

Also, what license to these rules have?

Comment (jaegeral) [reply data/windows/win_susp_add_sid_history.yml:1]: Clarified with the Sigma people. A link to the Sigma repo is good. The rules are all under: https://github.com/Neo23x0/sigma/blob/master/LICENSE.LGPL.txt

Comment (jaegeral) [reply data/windows/win_susp_add_sid_history.yml:1]: addressed

**Comment by jaegeral** [comment]
After talking with @berggren , we might want to wait with this PR before we have a proper way to ingest all rules from a remote git repo and maybe whitelist certain rules that are ready to be used.

**Comment by jaegeral** [comment]
This one is polluted, please look at: https://github.com/google/timesketch/pull/1231

**Pull request closed without merging**